nemo_curator/__init__.py

```diff
@@ -12,15 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import sys
+
 import dask
 
 # Disable query planning if possible
 # https://github.com/NVIDIA/NeMo-Curator/issues/73
-if dask.config.get("dataframe.query-planning") is True:
+if dask.config.get("dataframe.query-planning") is True or "dask-expr" in sys.modules:
```

Comment:
How about:

```python
query_planning_config = dask.config.get("dataframe.query-planning")
if query_planning_config is True or (
    query_planning_config is None and
    "dask.dataframe" in sys.modules and
    "dask_expr" in sys.modules
):
    # Cannot change query-planning config anymore, must raise
    raise NotImplementedError("...")
else:
    dask.config.set({"dataframe.query-planning": False})
```

This seems like the only way to check if you can still modify the query-planning config (pre https://github.com/dask/dask/pull/11183).

Reply:
These are somewhat contrived examples, but if the following happens:
```python
import dask
from dask import dataframe as dd

dask.config.set({"dataframe.query-planning": False})
import nemo_curator
```

```python
import dask
import dask_expr

dask.config.set({"dataframe.query-planning": False})
import nemo_curator
```

Both cases would not raise an error on the Curator side even though query planning is indeed enabled. Shouldn't the only criteria to check whether query planning might be used is checking the presence of `dask_expr` in sys.modules even if query_planning_config can be false?

Reply:
Sorry for the ridiculously reply!

>Shouldn't the only criteria to check whether query planning might be used is checking the presence of dask_expr in sys.modules

Okay - I thought about this again and agree with your preference to be as strict as possible. Although it is possible to use `dask_expr` separately from `dask.dataframe`, it seems very unlikely that a user will be doing this. Therefore, I agree that your original logic make sense.

With that said, it should become easier to check if query-planning is enabled after https://github.com/dask/dask/pull/11183 (curator can import `dask.dataframe` after setting the config, and raise an error if `dd.DASK_EXPR_ENABLED` is `True`).

Reply:
```suggestion
if dask.config.get("dataframe.query-planning") is True or "dask_expr" in sys.modules:
```